src/main/java/htsjdk/tribble/annotation/Strand.java

```diff
@@ -24,29 +24,116 @@
+     */
+    private final static Strand[] VALUES = values();
+
+    /**
+     * How we represent the strand as a single {@code char}.
+     */
+    private final char charEncoding;
+
+    /**
+     * How we represent the strand as a {@link String}.
+     */
+    private final String stringEncoding;
+
+    Strand(final char ch) {
+        charEncoding = ch;
+        stringEncoding = String.valueOf(charEncoding);
     }
 
     /**
      * provide a way to take an encoding string, and produce a Strand
      * @param encoding the encoding string
      * @return a Strand object, if an appropriate one cannot be located an IllegalArg exception
+     * @deprecated please use {@link #decode} instead.
+     */
+    @Deprecated
+    public static Strand toStrand(final String encoding) {
+        return decode(encoding);
+    }
+
+    /**
+     * Returns the {@link Strand} that a {@code char} value represents.
+     * @param ch the char encoding for a Strand.
+     * @throws IllegalArgumentException if input char does not
+     *   represent a {@link #Strand} instance.
+     * @return never {@code null}, a value so that {@code decode(c).encodeAsChar() == c}.
+     */
+    public static Strand decode(final char ch) {
+        for(final Strand value : VALUES) {
+            if (value.charEncoding == ch) {
+                return value;
+            }
+        }
+        throw new IllegalArgumentException("Unable to match encoding to Strand enum for encoding char: " + ch);
+    }
+
+    /**
+     * Returns the {@link Strand} that a {@link String} encodes for.
+     * @param encoding the strand string representation.
+     * @throws IllegalArgumentException if input string does not
+     *   represent a {@link #Strand} instance.
+     * @return never {@code null}, a value so that {@code decode(s).encode().equals(s)}.
      */
-    public static Strand toStrand(String encoding) {
-        for (Strand st : Strand.values())
-            if (st.encoding.equals(encoding))
-                return st;
-        throw new IllegalArgumentException("Unable to match encoding to Strand enum for encoding string " + encoding);
+    public static Strand decode(final String encoding) {
+        if (encoding != null && encoding.length() == 1) {
+            return decode(encoding.charAt(0));
+        }
+        throw new IllegalArgumentException("Unable to match encoding to Strand enum for " +
+                "encoding string: " + encoding);
     }
 
+    /**
+     * Returns a string representation of this {@link Strand}
+     * @return never {@code null}, a value so that {@code decode(encode(X)) == X}.
+     */
+    public String encode() {
+        return stringEncoding;
+    }
+
+    /**
+     * Returns a single char encoding of this {@link Strand}.
+     * @return a value so that {@code decode(encodeAsChar(X)) == X}.
+     */
+    public char encodeAsChar() {
+        return charEncoding;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Overriding this method makes conversion of  `String` <-> `Strand` failed in an unexpected way (unless documented). For example, `Strand.valueOf(Strand.POSITIVE.toString())` won't return `Strand.POSITIVE` anymore. Because `valueOf` cannot be overriden, I think that the `toString()` method shouldn't return the encoding....

Reply:
mmm... looking at the standard Java doc for ```name()```  it seems as the overriding of toString() is actually encouraged to provide a more user friendly string representation of the enum constant. 

Also valueOf javadoc says explicitly is mean to recover constant by its constant name so to be used like so 
```
String x = X.name();
...
// millions lines later in another class far far a way:
...
EnumX X = EnumX.valueOf(y) // where somehow y came from x.
```
So assuming that ```valueOf(X.toString()) == X``` is a understandable misinterpretation  of what ```valueOf``` is supposed to do. I guess that that is why it makes sense for it to fail with an Exception rather than return a null as go are getting wrong the hard-wired programatic name for a constant.

That said, it is true that this change break back-copatability a bit and I don't feel very strong about it ...to show "strand = POSITIVE" rather than "strand = +" in error messages doesn't really make much of a difference.... an further enforces the use of ```encode``` and ```decode``` for reading it in or out of files... so I will revert that. 

Any other opinions.

Reply:
You're perfectly right, sorry. It is the value of `X.name()` the one that should be kept the same. I'm OK with the change, even if it is not backwards compatible -  in principle I am in favor of `toString()` methods only implemented for error/debugging messages and not rely on them for other purposes...